business/clustering.go

```diff
@@ -0,0 +1,160 @@
+package business
+
+import (
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+)
+
+// ClusteringService is a support service for retrieving data about the mesh environment
+// when Istio is installed with multi-cluster enabled.
+type ClusteringService struct {
+	k8s           kubernetes.ClientInterface
+	businessLayer *Layer
+}
+
+// MeshCluster holds some metadata about a cluster that is
+// part of the mesh.
+type MeshCluster struct {
+	// ApiEndpoint is the URL where the Kubernetes/Cluster API Server can be contacted
+	ApiEndpoint string `json:"apiEndpoint"`
+
+	// IsHomeCluster specifies if this cluster is hosting Kiali (and the observed Mesh Control Plane)
+	IsHomeCluster bool `json:"isHomeCluster"`
```

Comment:
Maybe we could be explicit with this and a few other characterizations of the cluster, like HasKiali, HasControlPlane, HasDataPlane etc?

Reply:
Hmm... Naming is hard here...

So, in my env I have two clusters. Both can be marked as HasKiali, HasControlPlane and HasDataPlane, yet only one should be marked with "IsHomeCluster" and the one that gets this flag depends completely on which Kiali instance you are accessing...

So, all three variables HasKiali, HasControlPlane and HasDataPlane are good ideas to have, but this PR is still not finding this data, so I'll not add them yet (but I'll try to add in follow-ups). 

Focusing on "IsHomeCluster"... well... It could be named like `IsHostingThisKialiInstance`???? .... I'm now sure, I don't have ideas of a good naming here. May be the comment is what I should fix?

Reply:
I see your point.  A couple of ideas, isHomeKiali, isCurrentKiali, isActiveKiali, isSessionKiali.